### PR TITLE
Sample showing WLS running on AKS

### LINKED ISSUE #1801: Document using the Operator on Azure Kubernetes Service

### PR DESCRIPTION
**Pull request by edburns**
Original work done by https://github.com/galiacheng based on other samples.

```
On branch edburns-msft-1801-wls-aks Fixes #1801
copied:     kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/README.md -> docs-source/content/samples/simple/azure-kubernetes-service/_index.md

- Move content over from samples area.

modified:   kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/README.md

- Point to document in docs.

renamed:    kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/screenshot-deploy-test-app.png -> docs-source/content/samples/simple/azure-kubernetes-service/screenshot-deploy-test-app.png
renamed:    kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/screenshot-logs.png -> docs-source/content/samples/simple/azure-kubernetes-service/screenshot-logs.png

- Move images over.
```

### CODE REVIEW

File: docs-source/content/samples/simple/azure-kubernetes-service/_index.md [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+
+NAME                            TYPE           CLUSTER-IP     EXTERNAL-IP     PORT(S)          AGE
+domain1-cluster-1-external-lb   LoadBalancer   10.0.108.249   52.224.248.40   8001:32695/TCP   30m
+```
+
+In the example, the application address is: `http://52.224.248.40:8001/testwebapp`.
+
+The test application will list the server host and server IP on the page.
+
+## Access WebLogic Logs
```
Comment (rosemarymarano) [inline docs-source/content/samples/simple/azure-kubernetes-service/_index.md:771]: WebLogic -> WebLogic Server

**Comment by edburns** [comment]
Thanks, @rosemarymarano .  We will address these in the scope of this PR.

File: docs-source/content/samples/simple/azure-kubernetes-service/_index.md [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+az role assignment create --assignee <appId> --role Contributor
+```
+
+### Docker Hub
+
+You will need a Docker Hub account. If you don't have an existing account, please sign up for a new account at [DockerHub](https://hub.docker.com/). Note down your username, password and  email for Docker Hub. Because this guide uses a Docker image for a specific version of WLS, and WLS requires accepting license terms, do a Docker Hub "checkout" of [Oracle WebLogic Server](https://hub.docker.com/_/oracle-weblogic-server-12c).  This guide was written with 12.2.1.3, but other versions may work as well.
+
+### Clone WebLogic Operator Repository
+
+Please clone this repository to your machine. We will use several scripts in this repository to create domain.
```
[the remark this answers is not in the document]
Comment (rosemarymarano) [reply docs-source/content/samples/simple/azure-kubernetes-service/_index.md:104]: to create domain -> to create the domain (you are missing the word "the.")

File: docs-source/content/samples/simple/azure-kubernetes-service/_index.md [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+
+```bash
+git clone https://github.com/oracle/weblogic-kubernetes-operator.git
+```
+  
+*The following sections of the guide will take you step-by-step through the process of setting up a WebLogic cluster on AKS - remaining as close as possible to a native Kubernetes experience. This allows you to understand and customize each step. If you wish to have a more automated experience that abstracts some lower level details, you can skip to the [Automation](#automation) section.*
+
+## Create the AKS Cluster
+
+This guide requires we disable http-application-routing by default.  If you want to enable  http_application_routing, please follow [HTTP application routing](https://docs.microsoft.com/azure/aks/http-application-routing).
```
[the remark this answers is not in the document]
Comment (rosemarymarano) [reply docs-source/content/samples/simple/azure-kubernetes-service/_index.md]: guide -> sample (the idea was to change this terminology globally)

File: docs-source/content/samples/simple/azure-kubernetes-service/_index.md [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+
+We will use the `kubernetes/samples/scripts/create-kuberetes-secrets/create-azure-storage-credentials-secret.sh` script to create the storage account key as a Kubernetes secret, leave secret name with default value `azure-secret`.  For example:
+
+```bash
+./create-azure-storage-credentials-secret.sh -a $AKS_PERS_STORAGE_ACCOUNT_NAME -k $STORAGE_KEY
+```
+
+## Generate Configuration Files
+
+This guide uses Kubernetes Persistent Volume Claims (PVC) and load balancing to bring WLS to AKS.  These features are expressed to Kubernetes using yaml files.  The script `kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks.sh` generates the required configuration files automatically, given an input file containing the parameters.  A parameters file is provided at `kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.yaml`.  Copy and customize this file for your needs.
```
[the remark this answers is not in the document]
Comment (rosemarymarano) [reply docs-source/content/samples/simple/azure-kubernetes-service/_index.md]: YAML is an acronym (YAML Ain't Markup Language) and should always be capitalized.

File: docs-source/content/samples/simple/azure-kubernetes-service/_index.md [hunk 5]
```diff
@@ -343,7 +344,7 @@ You should see this output:
 clusterrolebinding.rbac.authorization.k8s.io/helm-user-cluster-admin-role created
 ```
 
-Install the WebLogic Operator. The Operator’s Helm chart is located in the `kubernetes/charts/weblogic-operator` directory. Please check the Helm version first if you are using the Azure Cloud Shell, and run the corresponding command.
+Install the WebLogic Server Operator. The operator’s Helm chart is located in the `kubernetes/charts/weblogic-operator` directory. Please check the Helm version first if you are using the Azure Cloud Shell, and run the corresponding command.
```
Comment (rosemarymarano) [inline docs-source/content/samples/simple/azure-kubernetes-service/_index.md:347]: WebLogic Server Operator -> WebLogic Server Kubernetes Operator (product name)

File: docs-source/content/samples/simple/azure-kubernetes-service/_index.md [hunk 6]
```diff
@@ -0,0 +1,853 @@
...
+   regcred                                   kubernetes.io/dockerconfigjson        1      2d20h
+   sh.helm.release.v1.weblogic-operator.v1   helm.sh/release.v1                    1      2d21h
+   weblogic-operator-secrets                 Opaque                                1      2d21h
+   ```
+
+   In the `NAME` column in your output is missing any of the values shown above, please reexamine your execution of the preceding steps in this guide to ensure you correctly followed all of them.  The `default-token-mwdj8` shown above will have a different ending in your output.
+
+3. We will use `kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/create-domain.sh` to create the domain in the persistent volume we created previously.
+
+   First, we need to set up domain configuration for the WebLogic domain.  This step uses the configuration generated previously.
```
[the remark this answers is not in the document]
Comment (rosemarymarano) [reply docs-source/content/samples/simple/azure-kubernetes-service/_index.md:439]: set up domain configuration -> set up the domain configuration (you are missing the word "the.")

**Comment by edburns** [comment]
Hello @galiacheng , while reviewing this with @m-reza-rahman , Reza requested the following change.

Reza found the use of heredoc to grant the `cluster-admin` role was inconsistent.  I realize this is handled automatically in the `function installWebLogicOperator {` and that's fine, but for the reader manually executing the steps in `_index.md` he felt it was jarring.  He wants the rbac grant YAML generated as one of the files output when the user runs this command.

```bash
./create-domain-on-aks.sh -i my-create-domain-on-aks-inputs.yaml -o ~/azure
```

* Modify the script to output this yaml file.

* Modify the `_index.md` in the following ways.

The output would look like

```bash
The following files were generated:
  /home/username/azure/weblogic-on-aks/pv.yaml
  /home/username/azure/weblogic-on-aks/pvc.yaml
  /home/username/azure/weblogic-on-aks/admin-lb.yaml
  /home/username/azure/weblogic-on-aks/cluster-lb.yaml
  /home/username/azure/weblogic-on-aks/domain1.yaml
  /home/username/azure/weblogic-on-aks/cluster-admin-role.yaml

Completed
```

Then, add a new paragraph at the very beginning of the `Apply generated
configuration files` section directing the reader to apply that file.
Consider this text as a suggestion.

-----------------------

Kubernetes Operators use [Helm](https://helm.sh/) to manage Kubernetes applications. You have to grant the Helm service account with the `cluster-admin` role with the following command.

```bash
kubectl apply -f ~/weblogic-on-aks/cluster-admin-role.yaml
```

You should see this output:

```bash
clusterrolebinding.rbac.authorization.k8s.io/helm-user-cluster-admin-role created
```

-----------------------------

Then, you must delete the corresponding text in the `Install WebLogic Operator` section, starting with "Kubernetes Operators use" and ending with removing the output section.

Also delete the sentence "Install the WebLogic Server Operator." because it is now unnecessary.

**Comment by galiacheng** [comment]
Hello @m-reza-rahman, I changed the `[operator documentation](/)` to `[operator documentation]({{< relref "/userguide/_index.md" >}})`, I got the url from docs-source\content\_index.md.

**Comment by edburns** [comment]
Closing this pull request to rework it.  Will open a new one.

**Pull request closed without merging**